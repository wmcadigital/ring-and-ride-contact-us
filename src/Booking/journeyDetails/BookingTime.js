import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import Question from "../../common/Question";
import ProgressIndicator from "../../common/ProgressIndicator";
import TimeInput from "../../common/TimeInput";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutJourneySectionName from "./getAboutJourneySectionName";

const BookingTime = () => {
  const stateApi = useFormState();
  const formValues = stateApi.values;
  const bookingParty = formValues["bookingParty"];

  let question = "";
  if (bookingParty === "mySelf") {
    question = "What time do you want to be picked up?";
  } else if (bookingParty === "behalfSomeone") {
    question = `What time does ${formValues["firstName"]} want to be picked up?`;
  } else if (bookingParty === "behalfGroup") {
    question = "What time is the group to be picked up?";
  }

  const error = stateApi.submitFailed
    ? stateApi.errors?.["outwardPickup.timeInput"]
    : null;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("2", stateApi)}
        sectionName={getAboutJourneySectionName(stateApi)}
      />
      <Question text={question} />
      <TimeInput label="Time of travel" prefix="outwardPickup" error={error} />
    </FormSection>
  );
};

export default BookingTime;
