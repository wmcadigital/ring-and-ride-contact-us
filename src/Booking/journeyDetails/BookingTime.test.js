import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import BookingTime from "./BookingTime";

jest.mock("react-final-form");

describe("BookingTime", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingTime />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfSomeone",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingTime />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for a group", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfGroup",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingTime />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
        firstName: "John",
      },
      submitFailed: true,
      errors: { "outwardPickup.timeInput": "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingTime />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
