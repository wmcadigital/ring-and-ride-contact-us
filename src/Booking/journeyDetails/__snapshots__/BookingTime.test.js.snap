// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingTime individual is filling out form for a group 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their journey"
    sectionPosition="Section 3 of 5"
  />
  <Question
    text="What time is the group to be picked up?"
  />
  <TimeInput
    error={null}
    label="Time of travel"
    prefix="outwardPickup"
  />
</FormSection>
`;

exports[`BookingTime individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="What time does John want to be picked up?"
  />
  <TimeInput
    error={null}
    label="Time of travel"
    prefix="outwardPickup"
  />
</FormSection>
`;

exports[`BookingTime individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="What time do you want to be picked up?"
  />
  <TimeInput
    error={null}
    label="Time of travel"
    prefix="outwardPickup"
  />
</FormSection>
`;

exports[`BookingTime renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="What time do you want to be picked up?"
  />
  <TimeInput
    error="Required"
    label="Time of travel"
    prefix="outwardPickup"
  />
</FormSection>
`;
